src/futures/update_stream.rs

```diff
@@ -257,4 +261,22 @@ mod tests {
     assert_eq!(COUNT_PUBLISHED.load(Ordering::Acquire), 2);
     assert_eq!(COUNT_COMMITTER.load(Ordering::Acquire), 2);
   }
+
+  /// Try to publish, leave it incomplete, then on_block_commit, finishes it,
+  /// test that the publishing event is properly reset and try_publish is not retried.
+  #[allow(dead_code)]
+  fn partial_publishing_then_eager_publishing() {}
+
+  ///it is publishing time, publishing doesn't finish, on_block_commit runs, eager_publish also doesn't finish.
+  /// test that the publishing timer is properly reset.
+  #[allow(dead_code)]
+  fn publisher_timer_resets_after_timed_and_eager_publishing_fail() {}
+
+  /// publishing timer will activate just as on_block_commit is being executed.
+  /// Unluckily the tasks executed go in the following order,
+  /// on_block_commit -> publishing_timer -> updater -> committer
+  /// The miner is reset, publishing will be tried, a consensus won't be ready, the committer reesets publishing.
+  /// Relies on the consensus not finding a valid solution on time.
+  #[allow(dead_code)]
+  fn publisher_timer_slightly_ahead_of_committer_event() {}
```

Comment:
What's the purpose of these? Are they basically just TODOs?

Reply:
these are scenarios and edge-cases that I considered while implementing the feature but it was very hard to actually replicate the scenario. I wanted to leave some record and they are also valid TODO tests.